classification/test_sensor_data_classification.py

```diff
@@ -39,35 +92,114 @@ class TestSensorDataClassification(unittest.TestCase):
   Test classification results for sensor data. 
   """
 
+
   def testClassificationAccuracy(self):
 
     """
     Test classification accuracy for sensor data.
     """
 
-    for noiseAmplitude in WHITE_NOISE_AMPLITUDES:
-      expParams = ("RUNNING EXPERIMENT WITH PARAMS:\n"
-                   " * numRecords=%s\n"
-                   " * noiseAmplitude=%s\n"
-                   " * signalAmplitude=%s\n"
-                   " * signalMean=%s\n"
-                   " * signalPeriod=%s\n"
-                   ) % (NUM_RECORDS,
-                        noiseAmplitude,
-                        SIGNAL_AMPLITUDE,
-                        SIGNAL_MEAN,
-                        SIGNAL_PERIOD)
-      print expParams
-
-      network = configureNetwork(noiseAmplitude)
-      numCorrect, numTestRecords, predictionAccuracy = runNetwork(network,
-                                                                  NUM_RECORDS,
-                                                                  PARTITIONS)
-
-      if noiseAmplitude == 0:
-        self.assertEqual(predictionAccuracy, 100)
-      else:
-        self.assertNotEqual(predictionAccuracy, 100)
+    networkConfigurations = _generateNetworkConfigurations()
+
+    for networkConfig in networkConfigurations:
+      for noiseAmplitude in WHITE_NOISE_AMPLITUDES:
+        for signalMean in SIGNAL_MEANS:
+          for signalAmplitude in SIGNAL_AMPLITUDES:
+            for signalPeriod in SIGNAL_PERIODS:
+
+              sensorType = networkConfig[SENSOR_REGION_NAME]["type"]
+              spEnabled = networkConfig[SP_REGION_NAME]["enabled"]
+              tmEnabled = networkConfig[TM_REGION_NAME]["enabled"]
+              upEnabled = networkConfig[UP_REGION_NAME]["enabled"]
+              classifierType = networkConfig[CLASSIFIER_REGION_NAME]["type"]
+
+              expParams = ("RUNNING EXPERIMENT WITH PARAMS:\n"
+                           " * numRecords=%s\n"
+                           " * signalAmplitude=%s\n"
+                           " * signalMean=%s\n"
+                           " * signalPeriod=%s\n"
+                           " * noiseAmplitude=%s\n"
+                           " * sensorType=%s\n"
+                           " * spEnabled=%s\n"
+                           " * tmEnabled=%s\n"
+                           " * upEnabled=%s\n"
+                           " * classifierType=%s\n"
+                           ) % (NUM_RECORDS,
+                                signalAmplitude,
+                                signalMean,
+                                signalPeriod,
+                                noiseAmplitude,
+                                sensorType.split(".")[1],
+                                spEnabled,
+                                tmEnabled,
+                                upEnabled,
+                                classifierType.split(".")[1])
+              print expParams
+
+              inputFile = generateData(DATA_DIR,
```

Comment:
Better to store this as `self.inputFile` so the `tearDown()` can be more explicit -- i.e. not guess what the file name is. This way if the file naming scheme in `generateData()` changes you won't  have to remember to change it here too.